### DISCUSSION TITLE
Gifsave removing semi-transparent colors

### DISCUSSION THREAD

**Discussion by jfcalvo** [libvips/libvips#2604] [Help wanted]
I'm moving from `magicksave` to new `gifsave` for saving GIF files and I have found a small difference in the behavior of these two functions.

Using this PNG file as input (`test.png`):

![test](https://user-images.githubusercontent.com/446497/147835804-d481a8c6-a9cc-4798-84f0-4c62f0d4917a.png)

I get the following GIF output using `magicksave` with `$ vips magicksave test.png test_magicksave.gif`:

![test_magicksave](https://user-images.githubusercontent.com/446497/147835879-37647f01-2e08-4361-8b01-60ea73bc13e3.gif)

Where we can see that the semi-transparent red color has the alpha removed and the full transparent center is fully transparent too.

But using `gifsave` I have a different output with `$ vips gif save test.png test_gifsave.gif`:

![test_gifsave](https://user-images.githubusercontent.com/446497/147835941-b6ec1ebd-3c3d-413f-960b-e728365375be.gif)

Where we can see the semi-transparent red color totally removed.

Do we have any way to have the behavior of the old `magicksave` where semi-transparent colors are not removed from the output image using the new `gifsave`?

I'm using vips 8.12.1.

Thanks.

**Comment by jcupitt** [comment]
Hi @jfcalvo,

That's an interesting test image. The semi-transparent red area is [255, 0, 0, 128], which `magicksave` was interpreting as opaque, but which the new GIF saver is interpreting as transparent.

Strictly I think 0-127 should be transparent and 128-255 should be opaque, so you're right, `gifsave` is wrong here. Let's change it for 8.12.2.

As a workaround for now, you could change the alpha threshold yourself. Extract the alpha, threshold to 0-128, and reattach.

**Reply by jcupitt** [reply]
Change in 8.12: https://github.com/libvips/libvips/commit/f75b3c1ee96fc0d7723034a4cf1d85855213d48b

**Reply by jfcalvo** [reply]
I will try your suggestion about changing the alpha threshold (or I can wait for version 8.12.2 too).

Thank you very much for the help and I wish you a happy new year! 🥳
